### ISSUE TITLE
Provide information on Neos Network Traffic

### ISSUE THREAD

**Issue by ProbablePrime** [Neos-Metaverse/NeosPublic#2611]
## Where have you searched for this information before making this issue?
Discord, Wiki, Even our Source. But its a little hard to find everything.

## Which part of the system do you need information on?
I have some friends who are either behind painful firewalls or are very tight on their home network security.

They'd like to understand how Neos operates on a network level in order to make sure that their firewalls can be configured appropriately.

## What information do you need to know?
1. A list of ports and protocols neos connects to
	1. For HTTP this should include domains, ports are somewhat standard here
2. Details of the NAT Punchthrough, I'm aware we use https://github.com/RevenantX/LiteNetLib but I'm not entirely sure what type of NAT Stuff we're doing. Does it follow a standard we can read about?
3. Any additional information that may help here.

## Do you plan to format and contribute the provided information to Wiki?
Sure, I'll document it.

It would be good to have this information available for those who are apprehensive of Neos etc as well so I think its a good thing to have :)

**Comment by Frooxius** [comment]
There's a few things:

- Cloud API
-- Most of it is at api.neos.com
-- SignalR makes a WebSocket connection to the Microsoft managed service. I don't have URL associated with that
-- Assets and other blobs are downloaded from CDN at https://cloudx2.azureedge.net
-- Thumbnails at https://cloudxthumbnails.azureedge.net/
-- Some more data might be accessed at https://cloudxstorage.blob.core.windows.net/

- LNL Realtime Networking
-- LNL uses UDP. Ports will vary, can be any port depending on host or when hosting (it lets OS pick a free port)
-- More info on NAT Punchthrough, it's generally implementation of this: https://en.wikipedia.org/wiki/UDP_hole_punching
-- NAT Punchthrough sends regular UDP packets to matchx.centralus.cloudapp.azure.com at 12500 port
-- LNL relay is the same as above but port 12600
-- Actual connection to the user can be obviously any IP or port

- Steam Networking Sockets realtime networking
-- This is completely relayed by the Steam client, we do not make any direct networking there. Unfortunately I don't know which exact ports and IP's this uses as that's on Valve's end

Generally speaking you shouldn't need to configure firewall in most cases, as that's mostly for inbound connections, rather than outbound. If you're configuring outbound, you'll run into a bit of a trouble as you won't be able to connect to anyone directly. It'll probably make it fall back to the relay, so the network performance might suffer.

When you host, Neos will start the session on a port randomly picked by the OS and listen in on the UDP socket for incoming connections. You can override the port when starting session or in the headless if you want to customize it and open it directly.

In layman terms, the NAT Punchthrough is sort of a way to make both ends think they're making outbound connection to each other at the same time with a help of a third party server (the one we host). With most NAT configurations that ends up working and a direct connection is established between the two parties.

It's like each side things they're the one who connected to the other, so the connection from their perspective is outbound rather than inbound and they're just getting data back from connection they (think) initiated.

Hope this helps!

**Comment by ProbablePrime** [comment]
Will transfer to wiki and then close.

Thank you!!

**Comment by ProbablePrime** [comment]
https://wiki.neos.com/Networking_Information created
